specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-03-31-privatepreview/PrivateDnsZone.json

```diff
@@ -0,0 +1,98 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-03-31-privatepreview",
+    "title": "PostgreSQLManagementClient",
+    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure PostgreSQL resources including servers, databases, firewall rules, VNET rules, security alert policies, log files and configurations with new business model."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBForPostgreSql/getPrivateDnsZoneSuffix": {
+      "post": {
+        "tags": [
+          "GetPrivateDnsZoneSuffix"
+        ],
+        "operationId": "GetPrivateDnsZoneSuffix_Execute",
+        "x-ms-examples": {
+          "GetPrivateDnsZoneSuffix": {
+            "$ref": "./examples/GetPrivateDnsZoneSuffix.json"
+          }
+        },
+        "description": "Get private DNS zone suffix in the cloud",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "schema": {
+              "type": "string"
+            },
+            "description": "The required parameters for getting private DNS zone suffix based on server type."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PrivateDnsZoneSuffix"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PrivateDnsZoneSuffix": {
+      "type": "string",
```

Comment:
>string [](http://example.com/codeflow?start=15&length=6)

You should consider modeling the response as an object because it will allow you to add more properties in the future without breaking changes.

Reply:
As discussed offline, the intention of this API is to return a dns zone suffix which is a string. We won't be adding anything to the response body.